frontend/src/static/js/components/webstatus-form-date-range-picker.ts

```diff
@@ -0,0 +1,204 @@
+
+  @query('#start-date')
+  startDateEl?: SlInput;
+
+  @query('#end-date')
+  endDateEl?: SlInput;
+
+  @query('#date-range-picker-btn')
+  submitBtn?: SlButton;
+
+  @state()
+  private _pendingStartDate = false;
+
+  @state()
+  private _pendingEndDate = false;
+
+  isValidDate(d: Date): boolean {
+    return !isNaN(d.getTime());
+  }
+
+  toIsoDate(date: Date): string {
+    return date.toISOString().slice(0, 10);
+  }
+
+  static get styles(): CSSResultGroup {
+    return [
+      SHARED_STYLES,
+      css`
+        .hbox,
+        .vbox {
+          gap: var(--content-padding-large);
+        }
+        #date-range-picker-btn {
+          justify-content: center;
+          margin-top: var(--sl-input-label-font-size-medium);
+        }
+      `,
+    ];
+  }
+
+  showPicker(input?: SlInput) {
+    input?.showPicker();
+  }
+
+  async handleStartDateChange(_: SlChangeEvent) {
+    const currentStartDate = this.startDate;
+    const newStartDate = new Date(this.startDateEl?.valueAsDate || '');
+    if (
+      !this.isValidDate(newStartDate) ||
+      this.minimumDate > newStartDate ||
+      this.endDate < newStartDate
+    ) {
+      this.startDateEl?.setCustomValidity(
+        `Date range should be ${this.toIsoDate(this.minimumDate)} to ${this.toIsoDate(this.endDate)} inclusive`,
+      );
+      this.startDateEl?.reportValidity();
+      this._pendingStartDate = false;
+      return;
+    }
+    if (newStartDate.getTime() !== currentStartDate.getTime()) {
+      this.startDateEl?.setCustomValidity('');
+      this.startDateEl?.reportValidity();
+      this.startDate = newStartDate;
+      this._pendingStartDate = true;
+    }
+  }
+
+  async handleEndDateChange(_: SlInputEvent) {
+    const currentEndDate = this.endDate;
+    const newEndDate = new Date(this.endDateEl?.valueAsDate || '');
+    if (
+      !this.isValidDate(newEndDate) ||
+      this.startDate > newEndDate ||
+      this.maximumDate < newEndDate
+    ) {
+      this.endDateEl?.setCustomValidity(
+        `Date range should be ${this.toIsoDate(this.startDate)} to ${this.toIsoDate(this.maximumDate)} inclusive`,
+      );
+      this.endDateEl?.reportValidity();
+      this._pendingEndDate = false;
+      return;
+    }
+    if (newEndDate.getTime() !== currentEndDate.getTime()) {
+      this.endDateEl?.setCustomValidity('');
+      this.endDateEl?.reportValidity();
+      this.endDate = newEndDate;
+      this._pendingEndDate = true;
+    }
+  }
+
+  handleSubmit() {
+    // Reset pending flags
+    this._pendingStartDate = false;
+    this._pendingEndDate = false;
+
+    // Dispatch a single event with both dates
+    const event = new CustomEvent<DateRangeChangeEvent>(
+      'webstatus-date-range-change',
+      {
+        detail: {
+          startDate: this.startDate,
+          endDate: this.endDate,
+        },
+      },
+    );
+    this.dispatchEvent(event);
+  }
+
+  isSubmitButtonEnabled() {
+    return this._pendingStartDate || this._pendingEndDate;
```

Comment:
Does the `_pending` prefix here mean that the dates exist and are valid?

Reply:
Yes, it means there are new dates that have been validated. I can add a note about this.